### DISCUSSION TITLE
Whats does the _NB suffix in enums mean

### DISCUSSION THREAD

**Discussion by JoHaHu** [official-stockfish/Stockfish#4335] [Q&A] [answered]
Almost all enums in [types.h](https://github.com/official-stockfish/Stockfish/blob/master/src/types.h) including a variant with _NB Suffix.

In the code those variants are mostly used to define the length of arrays.
1. Is _NB an appreviation for something i cant figure out? (Only thing i can thing of is 'Number' or 'Number of Bits', but that doesn't look right)
2. There are some of those bigger than they might need to be e.g. PIECE_NB. Because they are all power of 2 i assume that they are chosen to have better memory alignment?

**Comment by vondele** [comment, marked as answer]
NumBer is right. Basically, gives you the NumBer of pieces, squares etc in the enum
